src/Psalm/Context.php

```diff
@@ -667,7 +753,7 @@ public function removeDescendents(
 
         foreach ($this->vars_in_scope as $var_id => $type) {
             if (preg_match('/' . preg_quote($remove_var_id, '/') . '[\]\[\-]/', $var_id)) {
-                unset($this->vars_in_scope[$var_id]);
+                $this->remove($var_id, false);
```

Comment:
Not sure why this wasn't unsetting `$this->vars_possibly_in_scope` before, but it works fine this way. I'm guessing it was just overlooked? Or maybe `$this->vars_possibly_in_scope` is guaranteed to not contain descendents? (side note, that should actually be descendant, descendent is an adjective with a subtly different meaning).

Reply:
well, vars_possibly_in_scope is just an array of bools. It's nothing without vars_in_scope so yeah, probably overlooked and never caused issue